cpp/daal/src/sycl/reducer.cpp

```diff
@@ -63,23 +64,37 @@ services::Status Reducer::buildProgram(ClKernelFactoryIface & kernelFactory, con
     return status;
 }
 
-void Reducer::singlepass(ExecutionContextIface & context, ClKernelFactoryIface & kernelFactory, Layout vectorsLayout, const UniversalBuffer & vectors,
-                         uint32_t nVectors, uint32_t vectorSize, uint32_t workItemsPerGroup, UniversalBuffer & reduceRes, services::Status & status)
+services::Status Reducer::singlepass(ExecutionContextIface & context, ClKernelFactoryIface & kernelFactory, Layout vectorsLayout,
+                                     const UniversalBuffer & vectors, uint32_t nVectors, uint32_t vectorSize, uint32_t workItemsPerGroup,
+                                     UniversalBuffer & reduceRes)
 {
+    services::Status status;
     auto reduce_kernel = kernelFactory.getKernel("reduceSinglepass", status);
-    DAAL_CHECK_STATUS_RETURN_VOID_IF_FAIL(status);
+    DAAL_CHECK_STATUS_VAR(status);
+
+    // no need to check overflow for nVectors * vectorSize due to we already have buffer vectors of such size
+    if (vectors.type() == TypeIds::id<float>())
+    {
+        DAAL_ASSERT_UNIVERSAL_BUFFER(vectors, float, nVectors * vectorSize);
+        DAAL_ASSERT_UNIVERSAL_BUFFER(reduceRes, float, nVectors);
+    }
+    else
```

Comment:
Maybe to check that vector is double and report error if both not float and not double?